### ISSUE TITLE
Maintenance: Convert deprecated `set-output` command on Github Actions workflows

### ISSUE THREAD

**Issue by dreamorosi** [aws-powertools/powertools-lambda-typescript#1127]
## Summary

We need to convert all the calls to set-output commands to the new format: https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-output-parameter

## Why is this needed?

GH is deprecating set-output (no ETA yet): https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

## Which area does this relate to?

Automation

## Solution

N/A

**Comment by dreamorosi** [comment]
There's now an ETA:
> Starting 1st June 2023 workflows using save-state or set-output commands via stdout will fail with an error.

**Comment by dreamorosi** [comment]
In #1335 we have removed the `set-output` from some of the workflows.

Based on a quick manual scan, these workflows still use it and need updating:
- Label PR based on title ([example run](https://github.com/awslabs/aws-lambda-powertools-typescript/actions/runs/4280435735))
- On new PR ([example run](https://github.com/awslabs/aws-lambda-powertools-typescript/actions/runs/4280435719)) 
- On PR merge ([example run](https://github.com/awslabs/aws-lambda-powertools-typescript/actions/runs/4280435723))
- Measure package size ([example run](https://github.com/awslabs/aws-lambda-powertools-typescript/actions/runs/4104342528))

**Comment by dreamorosi** [comment]
The fix in the `reusable_export_pr_details` was applied in the Python repo and merged in: awslabs/aws-lambda-powertools-python#1957.

I have also opened a PR in the 3rd party action (flochaz/pkg-size-action#3) we use, to patch the `setOutput` calls made via the `@actions/core` sdk.

Action items:
- Apply the same changes as the PR in the Python repo
- Test the "Measure package size" workflow once the PR in the other repo is merged
